warehouse/query-core/src/main/java/datawave/query/iterator/QueryLogIterator.java

```diff
@@ -0,0 +1,164 @@
+package datawave.query.iterator;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.YieldCallback;
+import org.apache.log4j.Logger;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+
+import static datawave.query.iterator.QueryOptions.QUERY_ID;
+
+public class QueryLogIterator implements SortedKeyValueIterator<Key, Value>{
+
+    protected static final Logger log = Logger.getLogger(QueryLogIterator.class);
+
+    protected SortedKeyValueIterator<Key,Value> source;
+    protected SortedKeyValueIterator<Key,Value> sourceForDeepCopies;
+    protected Map<String,String> queryOptions;
+    protected NestedIterator<Key> initKeySource, seekKeySource;
+    protected IteratorEnvironment myEnvironment;
+
+    protected Range range;
+
+    protected Key key;
+    protected Value value;
+    protected YieldCallback<Key> yield;
+
+    public QueryLogIterator() { }
+
+    public QueryLogIterator(QueryLogIterator other, IteratorEnvironment env) {
+        this.source = other.source.deepCopy(env);
+        this.sourceForDeepCopies = source.deepCopy(env);
+        this.initKeySource = other.initKeySource;
+        this.seekKeySource = other.seekKeySource;
+        this.myEnvironment = other.myEnvironment;
+        this.queryOptions = other.queryOptions;
+
+    }
+    @Override
+    public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
+
+        try{
+            if (log.isInfoEnabled()) {
+                log.info("QueryLogIterator: init() Started QueryID: " + options.get(QUERY_ID));
+            }
+            this.queryOptions = options;
+            this.myEnvironment = env;
+            this.source = source;
+        }
+        finally{
+            if (log.isInfoEnabled()) {
```

Comment:
Perhaps an "end" method akin to the start method.  Also this method could be the one that resets the thread name to its original form.